tests/core/snap-set-core-config/task.yaml

```diff
@@ -134,3 +137,8 @@ execute: |
     snap unset system system.kernel.printk.console-loglevel
     not test -e /etc/sysctl.d/99-snapd.conf
     sysctl -n kernel.printk|MATCH "^4\s+"
+
+    echo "setting the hostname works"
+    hostname > current-hostname
```

Comment:
Totally optional nitpick: you could use `tests.cleanup defer hostnamectl set-hostname "$(hostname)"` and use `tests.clenaup {prepare,restore}` in appropriate places.